p2p/node/pubsubManager/utils.go

```diff
@@ -94,41 +94,52 @@ func NewTopic(genesis common.Hash, location common.Location, data interface{}) (
 	return t, nil
 }
 
-func TopicFromString(genesis common.Hash, topic string) (*Topic, error) {
+func TopicFromString(topic string) (*Topic, error) {
 	topicParts := strings.Split(topic, "/")
 	if len(topicParts) < 3 {
-		return nil, ErrMalformedTopic
+		return nil, errors.New("topic string should have 3 parts")
+	}
+	if len(topicParts[0]) != 66 {
```

Comment:
These checks would be nice to have as a function in utils so we can reuse it (if it doesn't already exist elsewhere in the codebase).

Reply:
I just opened a can of worms so nasty I regret ever writing this commnent

Reply:
the places where we would use it already handle these errors in their own way, so I don't see a need to do this :shrug: